pkg/api/dashboard_public_test.go

```diff
@@ -325,10 +329,14 @@ func TestAPIQueryPublicDashboard(t *testing.T) {
 	t.Run("Returns query data when feature toggle is enabled", func(t *testing.T) {
 		server, fakeDashboardService := setup(true)
 
+		fakeDashboardService.On("GetPublicDashboard", mock.Anything, mock.Anything).Return(&models.Dashboard{}, nil)
```

Comment:
Since we're making multiple calls in the api method, we need to mock underlying method calls as well. I think this is.. ok but referencing https://github.com/grafana/grafana/pull/50982/files#r900677737 I think we need to roll some of this logic up and refactor